tools/tests_wycheproof_generate_ecdh.py

```diff
@@ -0,0 +1,171 @@
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or https://www.opensource.org/licenses/mit-license.php.
+'''
+Generate a C file with ECDH testvectors from the Wycheproof project.
+'''
+
+import json
+import sys
+
+from binascii import hexlify, unhexlify
+from wycheproof_utils import to_c_array
+
+def should_skip_flags(test_vector_flags):
+    # skip these vectors because they are for ASN.1 encoding issues and other curves
+    flags_to_skip = {"InvalidAsn", "WrongCurve"}
+    return any(flag in test_vector_flags for flag in flags_to_skip)
+
+def should_skip_tcid(test_vector_tcid):
+    '''
+    We skip some test case IDs that have a public key whose custom ASN.1 representation explicitly
+    encodes some curve parameters that are invalid. libsecp256k1 never parses this part so we do
+    not care testing those. See https://github.com/bitcoin-core/secp256k1/pull/1492#discussion_r1572491546
+    '''
+    tcids_to_skip = [496, 497, 502, 503, 504, 505, 507]
+    return test_vector_tcid in tcids_to_skip
+
+# Rudimentary ASN.1 DER public key parser.
+# This should not be used for anything other than parsing Wycheproof test vectors.
+def parse_der_pk(s):
+    tag = s[0]
+    L = int(s[1])
+    offset = 0
+    if L & 0x80:
+        if L == 0x81:
+            L = int(s[2])
+            offset = 1
+        elif L == 0x82:
+            L = 256*int(s[2]) + int(s[3])
+            offset = 2
+        else:
+            raise ValueError("invalid L")
+    value = s[(offset+2):(L+2+offset)]
+    rest = s[(L+2+offset):]
+
+    if len(rest) > 0 or tag == 0x06: # OBJECT IDENTIFIER
+        return parse_der_pk(rest)
+    if tag == 0x03: # BIT STRING
+        return value
+    if tag == 0x30: # SEQUENCE
+        return parse_der_pk(value)
+    raise ValueError("unknown tag")
+
+def parse_public_key(pk):
+    der_pub_key = parse_der_pk(unhexlify(pk))   # Convert back to str and strip off the `0x`
+    return hexlify(der_pub_key).decode('utf-8')[2:]
+
+def normalize_private_key(sk):
+    # Ensure the private key is at most 64 characters long, retaining the last 64 if longer.
+    normalized = sk[-64:].zfill(64)
+    assert len(normalized) == 64, "private key must be exactly 64 characters long."
+    return normalized
+
+def normalize_expected_result(er):
+    result_mapping = {"invalid": 0, "valid": 1, "acceptable": 1}
+    return result_mapping[er]
+
+filename_input = sys.argv[1]
+
+with open(filename_input) as f:
+    doc = json.load(f)
+
+num_groups = len(doc['testGroups'])
+
+num_vectors = 0
+offset_sk_running, offset_pk_running, offset_shared = 0, 0, 0
+out = ""
+private_keys = ""
+shared_secrets = ""
+public_keys = ""
+cache_sks = {}
+cache_public_keys = {}
+
+for i in range(num_groups):
+    group = doc['testGroups'][i]
+    num_tests = len(group['tests'])
+    assert group["type"] == "EcdhTest"
+    assert group["curve"] == "secp256k1"
+    for j in range(num_tests):
+        test_vector = group['tests'][j]
```

Comment:
```suggestion
    for test_vector in group['tests']:
```

And then we can remove `num_tests = len(group['tests'])`